### PR TITLE
feat: memberRemoveCleanup

### PR DESCRIPTION
**Pull request by IsaacArcos248**
# Pull Request

## Related Issue

Closes # 837

## Documentation

For _any_ version updates, please verify if the [documentation page](https://www.beccalyria.com/discord-documentation) needs an update. If it does, please [create an issue there](https://github.com/BeccaLyria/discord-documentation/issues/new?assignees=nhcarrigan&labels=%F0%9F%9A%A6+status%3A+awaiting+triage&template=update.md&title=%5BUPDATE%5D) to ensure it is not forgotten.

- [ ] My contribution does NOT require a documentation update.
- [X] My contribution DOES require a documentation update.

### CODE REVIEW

File: src/modules/guild/memberRemoveCleanup.ts [hunk 1]
```diff
@@ -0,0 +1,36 @@
+import LevelModel from "../../database/models/LevelModel";
+import StarModel from "../../database/models/StarModel";
+import { BeccaInt } from "../../interfaces/BeccaInt";
+import { beccaErrorHandler } from "../../utils/beccaErrorHandler";
+
+export const memberRemoveCleanup = async (
+  Becca: BeccaInt,
+  userId: string,
+  guildId: string
+) => {
```
Comment (nhcarrigan) [inline src/modules/guild/memberRemoveCleanup.ts:10]:
```suggestion
/*
 * Module to clean up a user's level and star data when
 * they leave a guild. 
 *
 * @param {BeccaInt} Becca Becca's Discord instance.
 * @param {string} userId Discord ID of the user that left the guild.
 * @param {string} guildId Discord ID of the guild the user left.
 */
export const memberRemoveCleanup = async (
  Becca: BeccaInt,
  userId: string,
  guildId: string
): Promise<void> => {
```

File: src/events/memberEvents/memberRemove.ts [hunk 2]
```diff
@@ -46,6 +47,7 @@ export const memberRemove = async (
     goodbyeEmbed.setTimestamp();
 
     await sendWelcomeEmbed(Becca, guild, goodbyeEmbed);
+    await memberRemoveCleanup(Becca, member.id, guild.id);
```
Comment (nhcarrigan) [inline src/events/memberEvents/memberRemove.ts:50]:
```suggestion
    await memberRemoveCleanup(Becca, member.id, guild.id);
```

**Review by nhcarrigan — CHANGES_REQUESTED** [review]
We forgot the JSDoc comment which the linter requires 🙂

File: src/modules/guild/memberRemoveCleanup.ts [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+import { beccaErrorHandler } from "../../utils/beccaErrorHandler";
+
+/*
+ * Module to clean up a user's level and star data when
+ * they leave a guild. 
+ *
+ * @param {BeccaInt} Becca Becca's Discord instance.
+ * @param {string} userId Discord ID of the user that left the guild.
+ * @param {string} guildId Discord ID of the guild the user left.
+ */
```
Comment (nhcarrigan) [inline src/modules/guild/memberRemoveCleanup.ts:13]:
```suggestion
/**
 * Module to clean up a user's level and star data when
 * they leave a guild.
 *
 * @param {BeccaInt} Becca Becca's Discord instance.
 * @param {string} userId Discord ID of the user that left the guild.
 * @param {string} guildId Discord ID of the guild the user left.
 */
```

**Comment by IsaacArcos248** [comment]
can you pwetty pwease appwuv my pw!!11

**Review by nhcarrigan — APPROVED** [review]
Looks good to me!

**Comment by nhcarrigan** [comment]
## Congratulations

Congratulations on making your first pull request to our project. We have reviewed your contributions and are happy to accept them.

We hope you enjoyed the contribution process, and encourage you to watch for issues labelled `help wanted` for additional opportunities to contribute.

**Pull request merged**